components/script/dom/performance.rs

```diff
@@ -148,10 +148,19 @@ pub(crate) struct Performance {
     resource_timing_buffer_current_size: Cell<usize>,
     resource_timing_buffer_pending_full_event: Cell<bool>,
     resource_timing_secondary_entries: DomRefCell<VecDeque<DomRoot<PerformanceEntry>>>,
+    timing: Dom<PerformanceTiming>,
+    navigation: Dom<PerformanceNavigation>,
 }
 
 impl Performance {
-    fn new_inherited(time_origin: CrossProcessInstant) -> Performance {
+    fn time_origin_to_millis(time_origin: CrossProcessInstant) -> u64 {
+        (time_origin - CrossProcessInstant::epoch()).whole_milliseconds() as u64
+    }
+
+    fn new_inherited(global: &GlobalScope, time_origin: CrossProcessInstant) -> Performance {
+        let nav_start = Self::time_origin_to_millis(time_origin);
+        let timing = PerformanceTiming::new(global, nav_start);
+        let navigation = PerformanceNavigation::new(global, CanGc::note());
```

Comment:
Please add a `can_gc: CanGc` argument to this method so callers pass it in.
```suggestion
        let navigation = PerformanceNavigation::new(global, can_gc);
```